content/general/events/aws-lambda.textile

```diff
@@ -0,0 +1,163 @@
+
+For @presence@ events, there will be a @presence@ field, which will contain the "@Presence@":/realtime/presence#presence-message event.
+
+minimize. batched @message@ payload example:
+  ```[json]
+  {
+    "source": "channel.message",
+    "appId": "aBCdEf",
+    "channel": "channel-name",
+    "site": "eu-central-1-A",
+    "ruleId": "1-a2Bc",
+    "presence": [{
+      "id": "abCdEFgHIJ:1:0",
+      "clientId": "bob",
+      "connectionId": "Ab1CDE2FGh",
+      "timestamp": 1582270137276,
+      "data": "some data in the presence object",
+      "action": 4
+    }]
+  }
+  ```
+
+h3(#no-envelope-examples). Non-enveloped event payloads
+
+Non-enveloped messages contain a few additional headers beyond the "shared headers":/general/events#examples. These are:
+
+- x-ably-envelope-appid := the "app ID":https://support.ably.io/support/solutions/articles/3000063083 which the message came from
+- x-ably-envelope-channel := the Ably channel which the message came from
+- x-ably-envelope-rule-id := the Ably Reactor Rule ID which was activated to send this message
+- x-ably-envelope-site := the Ably datacenter which sent the message
+- x-ably-envelope-source := the "source":/general/events#sources for the WebHook, namely "@channel.message@" or "@channel.presence@"
+- x-ably-message-connection-id := the connection ID responsible for the initial event
+- x-ably-message-id: qfPeURsWJN:2:0
+- x-ably-message-timestamp: 1582270371499
+
+Each enveloped message will have the following fields:
+
+- source := the source for the WebHook, namely "@channel.presence@" or "@channel.message@"
+- appId := the Ably app this message came from
+- channel := the Ably channel where the event occured
+- site :=
+- timestamp := a timestamp represented as milliseconds since the epoch for the presence event
+
+In addition, it will contain another field which will contain the actual message, which is named according to the message type.
+
+h4(#no-envelope-example-message). Non-enveloped message events
+
+For @Message@ events, there will be the additional headers:
+
+- x-ably-message-name := The "name":/realtime/messages#name of the @Message@
+
+The payload will contain the "data":/rest/messages#data of the @Message@.
+
+For example, if you sent the following curl message, which sends a JSON message to the channel @my_channel@:
+
+```[curl]
+curl -X POST https://rest.ably.io/channels/my_channel/messages \
+          -u "{{API_KEY}}" \
+          -H "Content-Type: application/json" \
+          --data '{ "name": "publish", "data": "example" }'
+
+```
+
+The @x-ably-message-name@ header would be @publish@, and the payload would be @example@.
+
+h4(#no-envelope-example-presence). Non-enveloped presence events
+
+For @Presence@ events, there will be the additional headers:
+
+- x-ably-message-action := the action performed by the event (@update@, @enter@, @leave)
+- x-ably-message-client-id := the client ID of the connection which sent the presence event
```

Comment:
messages may also have a client-id if the message was published by an identified client